lua/nvim-treesitter/indent.lua

```diff
@@ -251,7 +282,14 @@ function M.get_indent(lnum)
             if c_is_last_in_line then
               -- If current line is outside the range of a node marked with `@aligned_indent`
               -- Then its indent level shouldn't be affected by `@aligned_indent` node
-              if c_srow and c_srow < lnum - 1 then
+              if c_srow and (c_srow < lnum - 1) then
+                indent = math.max(indent - indent_size, 0)
```

Comment:
I am still unsure of whether we should have math.max() here
My thoughts then were to have this as a "reset" button for going out of range, and not "reset back to at least 0" button. It often times go below 0 and then will get added back up to the correct value

Reply:
Agreed. It's an odd one and the ordering is strange because we deal with decent first but iirc it can go negative on some lines before being reindented. 

I need to look at some examples with the debug statements I have locally on. Thanks for pointing this out.